api/src/main/java/org/codice/ddf/admin/api/validation/SourceValidationUtils.java

```diff
@@ -71,4 +104,42 @@
         }
         return validationResults;
     }
+
+    public List<ConfigurationMessage> validateNonDuplicateSourceName(String sourceName,
+            String configFieldId) {
+        List<ConfigurationMessage> validationResults = validateString(sourceName, configFieldId);
+        if (validationResults.isEmpty()) {
+            List<Map<String, Map<String, Object>>> configurations = FACTORY_PIDS.stream()
```

Comment:
This won't work in a system where a developer has added another source type that this validation class doesn't know about.